jobclient.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// JobFetchClient fetches jobs from servers
+type JobFetchClient interface {
+	listJobs() (*appsResp, error)
+	listFinishedJobs(since time.Time) (*jobsResp, error)
+	fetchJobDetails(id string) (jobDetail, error)
+	fetchTasks(id string) (tasks, error)
+	fetchCounters(id string) ([]counter, error)
```

Comment:
could you articulate the difference between a `fetch*` and a `list*` operation? especially as this one appears to return something resembling a list…

Reply:
ah good point. i didnt change any names in this PR, just moved the funcs to a different file.

`listCounters` definitely makes sense. `fetchTasks` tho is a bit weird because it returns a single `tasks` struct? that one seems to be a deeper problem.

How about I update `fetchCounters` --> `listCounters` in this PR, and leave a todo comment to revisit `fetchTasks` in the future?